### ISSUE TITLE
Problems when using sudo with another command like rm with options

### ISSUE THREAD

**Issue by iandol** [rsteube/carapace-bin#985]
On macOS, using latest Elvish nightly (0.19.0-dev.5dc8c02a32) and carapace 0.10.2 if I use a command like `rm` with options directly, the path component completes without problems; but if I wrap it with a sudo then I get an error:

```
▶︎ rm -rf /Library/Apple/
 COMPLETING argument
```

```
▶︎ sudo rm -rf /Library/ERR
 COMPLETING argument
ERR (unknown shorthand flag: 'r' in -rf)  _
```

**Comment by rsteube** [comment]
Yes, that is not yet working as i wasn't sure yet how to approach this problem.
The issue is that `sudo` has flags as well and the way `pflag` works is that it tries to parse the whole command line eagerly:

```sh
sudo --flags --of --sudo cmd --flags --of --cmd
```
Normaly further flag parsing is prevented with the `--` argument but i assume `sudo` itself stops at the first positional argument (`cmd`).
So the error you see is the `sudo` completer complaining it does not know the flag of the `rm` command.

I think i have an idea now what i could do to get around this problem though.

**Comment by rsteube** [comment]
`sudo -- rm -rf /Library/` this should work btw., i think i had another command in mind that didn't support `--` (still not great user experience though).

**Comment by iandol** [comment]
Thanks, I'll use this workaround for the moment (I can assign an elvish abbreviation to auto-complete this), I understand it is hard to know when the options for sudo finish and those of the command begin!
